control-plane-operator/controllers/hostedcontrolplane/cloud/powervs/powervs.go

```diff
@@ -0,0 +1,207 @@
+accountID = {{.AccountID}}
+clusterID = {{.ClusterID}}
+g2workerServiceAccountID = {{.G2workerServiceAccountID}}
+g2Credentials = {{.G2Credentials}}
+g2ResourceGroupName = {{.G2ResourceGroupName}}
+g2VpcSubnetNames = {{.G2VpcSubnetNames}}
+g2VpcName = {{.G2VpcName}}
+region = {{.Region}}
+powerVSCloudInstanceID = {{.PowerVSCloudInstanceID}}
+powerVSRegion = {{.PowerVSRegion}}
+powerVSZone = {{.PowerVSZone}}`
+
+var ccmConfigTemplate = template.Must(template.New("ccmConfigMap").Parse(ccmConfigTemplateData))
+
+func ReconcileCCMConfigMap(ccmConfig *corev1.ConfigMap, hcp *hyperv1.HostedControlPlane) (err error) {
+	config := map[string]string{
+		"ConfigFile":               fmt.Sprintf("%s/kubeconfig", kubeConfigFileBasePath),
+		"AccountID":                hcp.Spec.Platform.PowerVS.AccountID,
+		"ClusterID":                hcp.Name,
+		"G2workerServiceAccountID": hcp.Spec.Platform.PowerVS.AccountID,
+		"G2Credentials":            fmt.Sprintf("%s/%s", secretMountPath, "ibmcloud_api_key"),
+		"G2ResourceGroupName":      hcp.Spec.Platform.PowerVS.ResourceGroup,
+		"G2VpcSubnetNames":         hcp.Spec.Platform.PowerVS.VPC.Subnet,
+		"G2VpcName":                hcp.Spec.Platform.PowerVS.VPC.Name,
+		"Region":                   hcp.Spec.Platform.PowerVS.VPC.Region,
+		"PowerVSCloudInstanceID":   hcp.Spec.Platform.PowerVS.ServiceInstanceID,
+		"PowerVSRegion":            hcp.Spec.Platform.PowerVS.Region,
+		"PowerVSZone":              hcp.Spec.Platform.PowerVS.Zone,
+	}
+
+	configData := &bytes.Buffer{}
+	ccmConfigTemplate.Execute(configData, config)
+
+	if ccmConfig.Data == nil {
+		ccmConfig.Data = map[string]string{}
+	}
+
+	ccmConfig.Data[ccmConfig.Name] = configData.String()
+
+	return
+}
+
+func ReconcileCCMDeployment(deployment *appsv1.Deployment, hcp *hyperv1.HostedControlPlane, ccmConfig *corev1.ConfigMap, releaseImage *releaseinfo.ReleaseImage) (err error) {
+	commandToExec := []string{
+		"/bin/ibm-cloud-controller-manager",
+		"--authentication-skip-lookup",
+		"--bind-address=$(POD_IP_ADDRESS)",
+		"--use-service-account-credentials=true",
+		"--configure-cloud-routes=false",
+		"--cloud-provider=ibm",
+		fmt.Sprintf("--cloud-config=%s/%s", ccmConfigMapMountPath, ccmConfig.Name),
+		"--profiling=false",
+		"--leader-elect=true",
+		"--leader-elect-lease-duration=137s",
+		"--leader-elect-renew-deadline=107s",
+		"--leader-elect-retry-period=26s",
+		"--tls-cipher-suites=TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256,TLS_AES_128_GCM_SHA256,TLS_CHACHA20_POLY1305_SHA256,TLS_AES_256_GCM_SHA384",
+		fmt.Sprintf("--kubeconfig=%s/kubeconfig", kubeConfigFileBasePath),
+		"--use-service-account-credentials=false",
+	}
+
+	deployment.Spec = appsv1.DeploymentSpec{
+		Replicas: &[]int32{1}[0],
+		Strategy: appsv1.DeploymentStrategy{Type: "Recreate"},
```

Comment:
Why Recreate when we use leader election?

Reply:
I have referred the properties from the asset https://github.com/openshift/cluster-cloud-controller-manager-operator/blob/master/pkg/cloud/ibm/assets/deployment.yaml 
@shalver @JoelSpeed please share your thoughts on this.
IMO we can use RollingUpdate strategy.

Reply:
On OCP, we use host networking to enable the CCM to initialise on the master before the CNI is set up. Because the pod binds to a host port, it then needs to be recreate else you may end up, during a rolling update, with a conflict in ports if the new pod is bound to the same node.

This may not be required for HyperShift assuming the CCM is being run as a pod within the virtual control plane.

In fact, the majority of the decisions we made for OCP to make this work during a bootstrap, probably aren't required here. I suggest a thorough review of the deployment and restrictions we impose to decide whether those are appropriate here

Reply:
Removed the strategy to leave it to default which is `RollingUpdate`